Sources/InteroperabilityTestsExecutable/InteroperabilityTestsExecutable.swift

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import ArgumentParser
+import GRPCCore
+import GRPCHTTP2Core
+import GRPCHTTP2TransportNIOPosix
+import NIOPosix
+
+@testable import InteroperabilityTests
+
+@main
+@available(macOS 14.0, iOS 17.0, watchOS 10.0, tvOS 17.0, *)
+struct InteroperabilityTestsExecutable {
+  public static func main(_ arguments: [String]?) async {
+    do {
+      var command = try parseAsRoot()
+      if var asyncCommand = command as? AsyncParsableCommand {
+        try await asyncCommand.run()
+      } else {
+        try command.run()
+      }
+    } catch {
+      exit(withError: error)
+    }
+  }
```

Comment:
This shouldn't be necessary?

Reply:
Oh, TIL. We were manually doing this in the v1 implementation but I guess it's not necessary anymore with the `@main` annotation. Nice.